Correct and Align Docstrings for User RPC Endpoints

karthikeyan24-kk opened the pull request:
## Summary
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, open one before creating this pull request. -->
Related to #4013

<!-- Concisely describe what the pull request does. -->

This PR updates the docstrings in the user RPC module to ensure they accurately match the function signatures and the actual return values.

The goal is to maintain correctness of the API documentation so developers using these RPC endpoints have reliable and accurate reference information.

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

karthikeyan24-kk committed:
Fix RPC user endpoint docstrings (#4013)

Anish9901 reviewed (changes requested):
This is not it, we are looking for fixing any:
- Inconsistencies in the names, order, of attributes specified in the docstrings w.r.t. the parameters of the relevant function.
- Inconsistencies in the attribute of the return specified in the docstrings w.r.t the actual return of the function.

The goal is to keep the API docs accurate for any uses/developer who wants to use these APIs. Rewriting the the attribute descriptions as done in this PR, adds little to no value to the aforementioned goal unless the description is wrong, in which case rewriting is acceptable.

Anish9901 commented:
Also, 
- PR description must include the "Developer Certificate of Origin" and should adhere to our PR template.
- If you only want to work on a subset of APIs, please change the PR description from `Fixes #4013` -> `Related to #4013`.

karthikeyan24-kk commented:
bro , i do not understand  what your saying can please point out what i
would to change


On Sat, 22 Nov 2025 at 03:35, Anish Umale ***@***.***> wrote:

> *Anish9901* left a comment (mathesar-foundation/mathesar#5001)
> <https://github.com/mathesar-foundation/mathesar/pull/5001#issuecomment-3564761309>
>
> Also,
>
>    - PR description must include the "Developer Certificate of Origin"
>    and should adhere to our PR template.
>    - If you only want to work on a subset of APIs, please change the PR
>    description from Fixes #4013 -> Related to #4013.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mathesar-foundation/mathesar/pull/5001#issuecomment-3564761309>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BXH3NWLXV2WJEMIPGK3TONT356EBRAVCNFSM6AAAAACM2PRKDCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTKNRUG43DCMZQHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

karthikeyan24-kk commented:
Hi @Anish9901, thank you for your detailed review.

* I have updated the PR according to your feedback:

* Restored the original descriptions where they were already correct.

* Focused only on fixing inconsistencies between docstrings and function signatures/return values.

* Ensured that no unnecessary rewrites were included.

* Updated the PR description to use “Related to #4013” instead of “Fixes #4013”.

karthikeyan24-kk committed:
Merge branch 'develop' into fix/rpc-docstrings

Anish9901 commented:
Thanks for your effort on this @karthikeyan24-kk, but, this is still not what we are looking for fixing this issue.
- This issue is mostly about checking and investigating if all the docstrings of our endpoints have the appropriate attributes that match function signatures so that the docs are accurate.
- This issue may or may not actually require changes to fix it.
- The types that you mentioned in the docstrings like `(int)` are automatically generated in our docs.
- Some of the rewrites e.g. `configured_role_id: The Django ID of the ConfiguredRole model instance for the collaborator.` -> `configured_role_id (int): The ID of the ConfiguredRole assigned to the collaborator.`, changes the semantics of the sentence entirely.
